lib/adf_diag.py

```diff
@@ -915,4 +933,25 @@ def setup_run_cvdp(self):
         print('For CVDP information visit: https://www.cesm.ucar.edu/working_groups/CVC/cvdp/')
         print('   ')
 
+    #########
+
+    def derive_variables(self,vars_to_derive=None,ts_dir=None):
+
+        """
+        Derive variables acccording to steps given here.  Since derivations will depend on the 
+        variable, each variable to derive will need its own set of steps below.
+        """
+
+        for var in vars_to_derive:
+            if var == "PRECT":
+                 # PRECT can be found by simply adding PRECL and PRECC
+                 # grab file names for the PRECL and PRECC files from the case ts directory
```

Comment:
We should probably have a check here that skips trying to calculate `PRECT` and prints out a warning if it can't find both `PRECC` and `PRECL`.

Reply:
Ok so the way I put in a check was as follows (I'm not sure this is the best way to do it but it seems pretty explicit and somewhat simple):
~~~
if len(glob.glob(os.path.join(ts_dir,"*PRECC*"))) and len(glob.glob(os.path.join(ts_dir,"*PRECL*"))):
    constit_files=sorted(glob.glob(os.path.join(ts_dir,"*PREC*")))
else:
    ermsg = "PRECC and PRECL were not both present; PRECT cannot be calculated."
    raise FileNotFoundError(ermsg)
~~~
What do you think of this? I like that it explicitly looks for both PRECC and PRECL, and I've tested and it works (including with failure if it doesn't find both files), just wondering if there's an even simpler way.